android/titanium/src/java/org/appcelerator/titanium/util/TiUIHelper.java

```diff
@@ -1221,4 +1222,64 @@ public static Uri getRedirectUri(Uri mUri) throws MalformedURLException, IOExcep
 		}
 		return mUri;
 	}
+
+	/**
+	 * Helper method for getting the actual color values for Views with defined custom backgrounds
+	 * that take advantage of color state lists.
+	 */
+	public static String getColorOfBackgroundForState(TiBackgroundDrawable backgroundDrawable, int[] state)
+	{
+		try {
+			// TiBackgroundDrawable's background can be either PaintDrawable or StateListDrawable.
+			// Handle the cases separately.
+			Drawable simpleDrawable = backgroundDrawable.getBackground();
+			if (simpleDrawable instanceof PaintDrawable) {
+				// For backwards compatibility return null if the required state is not the default one.
+				if (state != TiUIHelper.BACKGROUND_DEFAULT_STATE_1) {
+					return null;
+				}
+				return transcriptColorIntToString(((PaintDrawable) simpleDrawable).getPaint().getColor());
+			}
+			// Get the backgroundDrawable background as a StateListDrawable.
+			StateListDrawable stateListDrawable = ((StateListDrawable) simpleDrawable);
```

Comment:
We need to do an `instanceof` check on the `StateListDrawable` to avoid the issue mentioned here...
https://github.com/appcelerator/titanium_mobile/pull/10744